src/navigators/createBottomTabNavigator.js

```diff
@@ -49,6 +49,16 @@ class TabNavigationView extends React.PureComponent<Props, State> {
     return label;
   };
 
+  _getTestIDProps = ({ route, focused }) => {
+    const { descriptors } = this.props;
+    const descriptor = descriptors[route.key];
+    const options = descriptor.options;
+
+    return typeof options.tabBarTestIDProps === 'function'
```

Comment:
I think it'll be better to have 2 different options, `tabBarTestID` and `tabBarAccessibilityLabel`. Also we probably don't need it to be a function since you can use `navigation.isFocused()` to detect if the route is focused.

Reply:
Also probably we should set `testID` of value `routeName` and `accessibilityLabel` of value `title`  by default.

Reply:
I dont think we should 'debate' the best approach, lets make it work as it did before, then use the standard process for making changes to the api.  This worked fine a few versions back.

Reply:
In principle I agree with your points @satya164. I found the tabBarTestIDProps also a bit unexpected. And I don't see any downside to using the label as a fallback (maybe someone else does?). But if tabBarTestIDProps is replaced with the testID and accessibilityLabel properties existing code will break in a subtle way.
Projects that already have tabBarTestIDProps in place for accessibility and not for automated tests may not realise their accessibility 'broke'.

On the other hand this is 2.0, so if you want to put in breaking changes, now is a great time :-)

Reply:
It looks like label can be a string or a function returning a component. That complicates the fallback a bit.

Reply:
> I dont think we should 'debate' the best approach, lets make it work as it did before

I'm not debating the best approach. It's a new feature in this package and doesn't need to carry over the legacy burden. This is a different package and we have already changed some other APIs. You can use the deprecated tab navigators if you're not ready to upgrade your app.

Reply:
> It looks like label can be a string or a function returning a component. That complicates the fallback a bit.

Yeah, we can ignore it if it's not a `string` or `null` or a function returning those. We should use the existing `_getLabelText` method in `createTabNavigator` to get the label.

Also I added `getAccessibilityLabel` and `getTestID` props in `react-native-tab-view` too. So we can move these to `createTabNavigator` and support in both material and bottom tabs.

Reply:
Perhaps my comment was a bit harsh about the debate, I agree it is a 2.0 release to breaking changes are fine, and its good to find a better way, but my point is that removing testability all together was a mistake.  I found a work around in the mean time that sort of solves both issues, as long as you are willing to modify the tabBarIcon in the navigationOptions, you can shove an element in there with a testID so at least the tests can find and tap on the tabs again. Granted its not as elegant as the old way, but does the trick.

Also when doing this I found it interesting, that actually there are 2 tabs rendered in this library so you NEED to use the focus as part of the ID otherwise when you search for the element you find both the non focused and focused element, which seems a little odd to me.  I haven't looked at the implementation as to why there are 2 elements, rather than one with conditional formatting, but thought i'd point it out as a heads up for those that want to take this approach.

```navigationOptions: ({
            navigation,
        }) => ({
            // tabBarTestIDProps: { 
            //     testID: `maintabs-${navigation.state.routeName}`,
            //     accessibilityLabel: `maintabs-${navigation.state.routeName}`,
            // },
            tabBarIcon: ({
                focused,
                tintColor,
            }) => {
 return (<View 
                    testID={`maintabs-${navigation.state.routeName}-${focused?'Y':'N'}`}
 ..... ```

Reply:
I'm a bit late to this conversation, but is there any reason why this component can't just pass down `testID` and `accessibilityLabel`?